### ISSUE TITLE
Remove Space before deletes numbers typed with the number pad. 

### ISSUE THREAD

**Issue by Mammal4963** [psoukie/zipchord#221]
**Describe the bug**
After a chord has been entered, followed by typing a number or numbers from the numpad then entering a symbol within the locale set to "remove space before", zipchord will delete the last number typed.

When you instead type a series of individual letters or numbers from the number row this issue doesn't happen. 

I expect that the final digit typed should not be deleted and instead the symbol should just be printed to the screen. 

**Useful Details**
- Which settings do you have selected in the ZipChord settings?
"Restrict chords while typing., Smart Spaces (all checked)"

- If the output is wrong, what is the sequence of keys pressed, the dictionary entry, and what text appears on the screen?

Here are two examples, one will demonstrate the undesired behavior, and the other shows the desired behavior. 

NOTE: The digit below must be typed using the number pad for the bug to occur. If you type it using the number row, then the 8 will not be deleted. 

(the + is not typed, it is my way of communicating that the following string is typed after something. So in the first example you are pressing the "hv" keys together as a chord, then the number 8, then the "#" symbol.)

Undesired behavior: (the 8 gets deleted)
input: hv+8+#
output: have #

Desired behavior: (the # sign doesn't delete anything in this case, which is correct)
input: hv+e i g h t+#
output: have eight#

I've added a video attachment to illustrate the issue. 
https://github.com/user-attachments/assets/03904a64-c56b-464b-9a01-e2971cb4b89b

**Comment by psoukie** [comment]
Thank you for reporting this and taking the time to provide the details and the video. I did not realize that numerical pad numbers registering as different keypresses can cause issues. The fix should be by adding the keys such as `{Numpad0}{Numpad1}` etc. into a list of **All keys** in the Locale Settings  window. That's not very elegant, so I'll experiment with adding these by default. In the long run, I'd like to get to a point where the app automatically adjusts based on the selected keyboard settings. See [Keyboard and language settings](https://github.com/psoukie/zipchord/wiki/Keyboard-and-language-settings) to add the numpad keys as a workaround. I'll see if I can fix this later this week.

**Comment by Mammal4963** [comment]
Thank you very much for your reply. After reading your comment, and the link to the keyboard and language settings page, I figured out I actually had to add the numpad digits to the local settings like this in order for it to work as expected, just adding {Numpad0} by itself made it to where the numpad0 didn't send any input at all.

This works: 
{Numpad0:0}{Numpad1:1}{Numpad2:2}{Numpad3:3}{Numpad4:4}{Numpad5:5}{Numpad6:6}{Numpad7:7}{Numpad8:8}{Numpad9:9}

**Comment by Mammal4963** [comment]
yeah I think that makes a lot of sense. most people don't distinguish between numpad1 & 1, so when 1 is in the locale under all keys I think it makes sense for that to cover the numpad1 too. I think only making the distinction if the user adds {numpad1} specifically as you stated makes the most sense and is the way to go.

**Comment by psoukie** [comment]
UPDATE: I misspoke. The `:` functionality was broken in a previous version. I'll fix it, and will introduce correct functionality for Numpad numbers.

**Comment by psoukie** [comment]
This should be resolved in [ZipChord 2.6](https://github.com/psoukie/zipchord/releases/tag/v.2.6.0). ZipChord no longer relies on locale-defined special-key mappings like `{Numpad1:1}`

Printable numeric keypad keys now have built-in representations used directly in shortcut definitions: [Numpad reference](https://github.com/psoukie/zipchord/wiki/Keyboard-and-language-settings#numeric-keypad-keys)

Note that 2.6 also removes the older custom locale stand-ins for non-printing special keys, so existing NumPad shortcuts will need to be updated to the predefined keypad.

If you still see incorrect behavior in 2.6, please let me know and I can reopen/investigate it further.
